src/main.rs

```diff
@@ -22,6 +22,14 @@ use crate::config::{Channel, Config};
 
 const TARGET: &str = env!("TARGET");
 
+/// List of files that should not be pruned even if they're not referenced in the manifest.
+const AVOID_PRUNING_UNUSED_FILES: &[&str] = &[
+    // Source code tarballs, which are not distributed through rustup and thus are mistakenly
+    // marked as "unused" by build-manifest.
+    "rustc-{release}-src.tar.gz",
+    "rustc-{release}-src.tar.xz",
```

Comment:
I am a bit concerned by this to be honest, as I mentioned elsewhere, but I think it's okay for now. It might make sense for us to update the manifest format with a "installable = false" or something for rustup, and otherwise still include everything that's part of a release in the manifest.

Reply:
Followed up in https://github.com/rust-lang/promote-release/issues/22.